File: backend/internal/model/vo/violation.go
package vo

import (
	"time"
	"tiny-forum/internal/model/do"
)

// ViolationVO 违规记录视图对象（用于 API 响应）
type ViolationVO struct {
	ID             uint               `json:"id"`
	CreatedAt      time.Time          `json:"created_at"`
	UpdatedAt      time.Time          `json:"updated_at"`
	UserID         uint               `json:"user_id"`
	OperatorID     *uint              `json:"operator_id,omitempty"` // 操作人ID，可为空
	ViolationType  do.ViolationType   `json:"violation_type"`
	Reason         string             `json:"reason"`
	Source         do.VoilationSource `json:"source"`
	Status         do.ViolationStatus `json:"status"`
	PunishType     do.PunishType      `json:"punish_type,omitempty"`
	PunishExpireAt *time.Time         `json:"punish_expire_at,omitempty"`
	AppealStatus   do.AppealStatus    `json:"appeal_status"`
	AppealReason   string             `json:"appeal_reason,omitempty"`
	AppealTime     *time.Time         `json:"appeal_time,omitempty"`
	AppealResult   string             `json:"appeal_result,omitempty"`
	User           *UserAuthVO        `json:"user,omitempty"`
	Operator       *UserAuthVO        `json:"operator,omitempty"`
}

File: frontend/src/shared/api/modules/user/violation.ts
import apiClient from "../../client";
import { ApiResponse, PageRequest } from "../../types/basic.model";
import { UserStatsVO } from "../../types/user.model";

/** 用户违规 */
export const userViolationApi = {
  // 获取当前用户的违规列表
  listUserViolations: () =>
    apiClient.get<ApiResponse<ViolationVO[]>>("/users/me/violations"),

  /** 用户申诉 */
  appeal: (id: string, reason: string) =>
    apiClient.post<ApiResponse<null>>(`/users/me/violations/${id}/appeal`, {
      reason,
    }),

  // 获取单条违规详情
  getViolationDetail: (id: string) =>
    apiClient.get<ApiResponse<ViolationVO>>(`/users/me/violations/${id}`),
};

export interface ViolationVO {
  id: string;
  created_at: string;
  updated_at: string;
  user_id: string;
  operator_id: string;
  violation_type: string;
  reason: string;
  source: string;
  status: string;
  punish_type: string;
  punish_expire_at: string;
  appeal_status: string;
  appeal_reason: string;
  appeal_time: string;
  appeal_result: string;
}
